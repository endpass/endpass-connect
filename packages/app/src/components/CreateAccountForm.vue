<template>
  <v-frame> 
    <message>
      You have not any account. Create one and you will be logged in.
    </message>
    <form-controls>
      <v-button
        type="primary"
        data-test="submit-button"
        @click="emitRequest"
      >
        Create account
      </v-button>
      <v-button 
        data-test="cancel-button" 
        @click="emitCancel"
      >Close</v-button>
    </form-controls>
  </v-frame>  
</template>

<script>
import VFrame from './VFrame.vue';
import VInput from './VInput.vue';
import VButton from './VButton.vue';
import Message from './Message.vue';
import FormControls from './FormControls.vue';

export default {
  name: 'CreateAccountForm',

  data: () => ({
    email: '',
  }),

  methods: {
    emitRequest() {
      this.$emit('request');
    },

    emitCancel() {
      this.$emit('cancel');
    },
  },

  components: {
    VFrame,
    VButton,
    VInput,
    Message,
    FormControls,
  },
};
</script>
