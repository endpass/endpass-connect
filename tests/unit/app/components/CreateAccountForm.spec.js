import { shallowMount, mount } from '@vue/test-utils';
import CreateAccountForm from '@@/app/src/components/CreateAccountForm.vue';

describe('CreateAccountForm', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(CreateAccountForm, {
        propsData: {},
      });
    });

    it('should correctly render CreateAccountForm component', () => {
      expect(wrapper.name()).toBe('CreateAccountForm');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(CreateAccountForm, {
        propsData: {},
      });
    });

    describe('submit feature', () => {
      it('should emit request on click submit button by default', () => {
        wrapper.find('[data-test=submit-button]').trigger('click');

        expect(wrapper.emitted().request).toBeTruthy();
      });
    });

    describe('cancel feature', () => {
      it('should emit close on click cancel button by default', () => {
        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeTruthy();
      });

      it('should not emit close on click cancel button if closable is false', () => {
        wrapper.setProps({
          closable: false,
        });

        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeFalsy();
      });
    });
  });
});
