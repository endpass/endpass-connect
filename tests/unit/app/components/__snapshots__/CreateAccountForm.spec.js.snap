// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountForm render should correctly render CreateAccountForm component 1`] = `
<div>
  <message-stub>
    You have not any account. Create one and you will be logged in.
  </message-stub>
  <form-controls-stub>
    <v-button-stub type="primary" data-test="submit-button">
      Create account
    </v-button-stub>
    <v-button-stub data-test="cancel-button">Close</v-button-stub>
  </form-controls-stub>
</div>
`;
